include/zephyr/net/conn_mgr.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_CONN_MGR_H_
+#define ZEPHYR_INCLUDE_NET_CONN_MGR_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if defined(CONFIG_NET_CONNECTION_MANAGER)
+
+struct net_if;
+struct net_l2;
+
+/**
+ * @brief Resend either NET_L4_CONNECTED or NET_L4_DISCONNECTED depending on whether connectivity
+ * is currently available.
+ */
+void net_conn_mgr_resend_status(void);
```

Comment:
From the API user perspective, what is the purpose of that function?

Reply:
It is a legacy feature from the previous version of `conn_mgr`. It appears to be provided so that applications that set up their event listeners after conn_mgr initialized can manually request the latest connectivity state be resent.

I think it might be worth it to eventually phase this function out, but I consider that task beyond the scope of these modifications